cytosnake/utils/config_utils.py

```diff
@@ -131,3 +132,73 @@ def load_cytosnake_configs() -> dict:
     # gets absolute path to cytosnake configs and load the configs
     cytosnake_config_path = cyto_paths.get_cytosnake_config_path()
     return load_configs(cytosnake_config_path)
+
+
+def update_config(
+    config_file_path: str | pathlib.Path,
+    new_key: str,
+    new_value: str | Any,
+    update=False,
+) -> None:
+    """This updates config level in the upper level.
+
+    Parameters
+    ----------
+    key : str
+        key to add into the config
+    value : str | Number
+        Value to add to the given key.
+    update : bool
+        Update value to existing key
+
+    Raises
+    ------
+    ValueError
+        raised if the key value cannot be converted into a string
+    TypeError
+        raised if there is a type error with any parameter
+    FileNotFoundError
+        raised if the config file is not found
+    KeyError
+        raised if given key exists within the config file
+    """
+
+    # type checking
+    if not isinstance(config_file_path, pathlib.Path):
+        if isinstance(config_file_path, str):
+            config_file_path = pathlib.Path(config_file_path).resolve(strict=True)
+        raise TypeError(
+            "'config_file_path' must be str or pathlib.Path."
+            f"not: {type(config_file_path)}"
+        )
+    if not isinstance(new_key, str):
+        try:
+            new_key = str(new_key)
+        except Exception:
+            raise ValueError("Unable to convert key value into a string")
```

Comment:
Consider referencing the root `Exception` with the `ValueError` to enhance the trace.
```suggestion
        except Exception as exc:
            raise ValueError("Unable to convert key value into a string") from exc
```